distributed/utils_test.py

```diff
@@ -225,7 +227,21 @@ def get_ip():
 original_config = copy.deepcopy(dask.config.config)
 
 
+@contextmanager
+def clean_config():
+    """A contextmanager that ensures the dask config is reset to the original
+    state during interpreter startup upon exit.
+    """
+    with patch.dict(dask.config.config, original_config, clear=True):
+        yield
+
+
 def reset_config():
+    warnings.warn(
+        "This methods will be removed soon. Consider using the contextmanager "
+        "distributed.utils_test.clean_config instead",
+        DeprecationWarning,
+    )
     dask.config.config.clear()
     dask.config.config.update(copy.deepcopy(original_config))
```

Comment:
I consider this to be an anti-pattern and would suggest to remove this. `utils_test` is kind of a public API so we'll need to go through a deprecation cycle.